proto/pulumi/resource.proto

```diff
@@ -39,6 +39,8 @@ service ResourceMonitor {
     // Register an invoke transform for the stack
     rpc RegisterStackInvokeTransform(Callback) returns (google.protobuf.Empty) {}
 
+    rpc RegisterLifecycleHook(RegisterLifecycleHookRequest) returns (google.protobuf.Empty) {}
```

Comment:
In general I think docs for all proto additions would be great. If we're holding off until finalised implementations, fair enough; apologies in that case!